src/symbolize/gimli/elf.rs

```diff
@@ -43,6 +43,45 @@ impl Mapping {
         })
     }
 
+    /// On Android, shared objects can be loaded directly from a
+    /// ZIP archive. For example, an app may load a library from
+    /// `/data/app/com.example/base.apk!/lib/x86_64/mylib.so`
+    ///
+    /// `zip_offset` should be page-aligned; the dynamic linker
+    /// requires this when it loads libraries.
```

Comment:
```suggestion
    /// `zip_offset` must be page-aligned if not None; the dynamic linker
    /// requires this when it loads libraries.
```

Reply:
Just to clarify: this is not a hard requirement for safety or avoiding a crash (ref: [my other comment](https://github.com/rust-lang/backtrace-rs/pull/662#discussion_r1749037840)).

If `zip_offset` is unaligned, the `mmap` call will fail and `new_android` will give up on loading the library from the archive.

Reply:
I reworded the documentation to hopefully explain this better; let me know if it still needs changes